### DISCUSSION TITLE
Sonoff RF Bridge  Tasmota wiki page

### DISCUSSION THREAD

**Discussion by marksmanaz** [arendst/Tasmota#18070] [Q&A]
The website for converting B1 to B0 is dead (bbconv.hrbl.pl) and the .py file does not work either. Without being able to convert the B1 data to B0 the wiki is useless.

**Comment by barbudor** [comment]
I don't have the device and don't know how to use but I tried to update the python script to python3
Can you test it ?
If it works, I will commit it to Tasmota repo

[BitBucketConverter.py.txt](https://github.com/arendst/Tasmota/files/10843456/BitBucketConverter.py.txt)

Note: it now makes use of requests intead of pycurl

**Reply by marksmanaz** [reply]
Thank you for that but it doesn't work still with Python3 the way it is. I couldn't get requests working. 
I did spend some time last night and today on it myself. I did get it to work but I had to comment out all references to PIL or Pillow. Not sure why a function was trying to drawimage and create a file called "oscilloscope.jpg". 🤷‍♂️ The program works without it just as well. There were a couple of changes I had to make one with a deprecated input command and the other was an extra "{"  it was looking for when scanning lines in the file.
I also added the feature of having all "good" scans written to an output.txt file it will create in the same dir as the py file when executed. As long as it has permission to do so in that Dir. 
 Each B0 line will be preceded by the date and will append to the output.txt file on a new line. You will have to manually delete the file or lines in the file to clean it up if so desired. It's nice to have a log that I could copy and paste from if need be.
Here is the file I tweaked if someone else wants to clean it up more or better than I did but it will work as is.
Thanks,
Mark
[B1toB0.py.txt](https://github.com/arendst/Tasmota/files/10846958/B1toB0.py.txt)

**Comment by Jason2866** [comment]
@jziolkowski Is this your hosting?

**Reply by jziolkowski** [reply]
ah indeed. I'll have a look

**Comment by jumble5** [comment]
Just want to share this solution. It worked for me!

https://npm.runkit.com/b1-to-b0-convertor

Just key this in the command window. Change the B1 code below to your own.

var b1ToB0Convertor = require("b1-to-b0-convertor")
const convertB1toB0 = require('b1-to-b0-convertor');
console.log(convertB1toB0('AA B1 04 1360 0276 0122 1388 38192A192A1A1A19292A1A19292929292A19292A1929292A1A192A192A1A19292A1A1A1A1A1A1A1A192A1A1A1A1A1A1A1A1A1A1A1A192A192929292A1A19292A1A19 55'));
